### PR TITLE
Add FIPS test case: apache_nssfips.pm

### PR DESCRIPTION
**Pull request by QingmingSu**
Configure Apache server with NSS module, also enable NSSFips

### CODE REVIEW

**Comment by QingmingSu** [comment]
Successful test run on local openQA:
http://147.2.212.224/tests/392

**Comment by coolo** [comment]
IMO it would be worth it to run this test slightly different also without FIPS. I.e. have one apache_nss and one apache_nssfips sharing common code. What do you think?

**Comment by QingmingSu** [comment]
Yeah, it makes sense. Which test suite should be added to when the new non-fips case (i.e. apache_nss) ready?

**Comment by coolo** [comment]
add it where php5.pm is

**Comment by QingmingSu** [comment]
Got it, thanks.
I will create new case apache_nss and make it share most codes with apache_nssfips

**Comment by coolo** [comment]
thanks in advance

**Comment by QingmingSu** [comment]
Updates:
  - Created lib/apachetest.pm for common codes of apache setup.
  - Added new case apache_nss.pm to test apache with nss module.
  - Added new case apache_nssfips.pm to test apache nss module in FIPS mode.

File: lib/apachetest.pm [hunk 1]
```diff
@@ -0,0 +1,102 @@
+package apachetest;
+
+use base Exporter;
```
Comment (coolo) [inline lib/apachetest.pm]: please add a license here too. the rest looks convincing

**Comment by QingmingSu** [comment]
Oh, I forgot the license statement for the library file.
The copyright has been added now.

The test cases have been tested in my local openQA:
--For non-fips cases: apache_ssl and apache_nss
http://147.2.212.224/tests/410

--For fips cases: apache_ssl and apache_nssfips
http://147.2.212.224/tests/411

File: lib/apachetest.pm (hunk 1 above)
Comment (QingmingSu) [inline lib/apachetest.pm]: Updated to add the missing license statement.

**Pull request merged**